### DISCUSSION TITLE
Mutli Enum classes use in Query param for one field

### DISCUSSION THREAD

**Discussion by miladva** [fastapi/fastapi#10730] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from enum import Enum

class X(str, Enum):
    A = "A"
    B = "B"


class Y(str, Enum):
    C = "C"
    D = "D"
    E = "E"


XY = X | Y


class QueryParams:
    def __init__(
        self,
        xy: XY):
        self.xy = xy


@router.post("")
async def test(
    payment_params: QueryParams = Depends()):
    pass
```

### Description

This is my code but this code does not show in Swagger.
![Screenshot from 2023-11-29 10-40-19](https://github.com/tiangolo/fastapi/assets/76900115/709cf638-3778-4f42-ac07-2b127bb3abe5)

I have 2 dedicated Enum Classes.
I used Enum and I wanna show my dropdown in Swagger.
what should I do? 

### Operating System

Linux

### FastAPI Version

0.104.1

### Pydantic Version

2.5.2

### Python Version

 3.11.6

**Comment by yinziyan1206** [comment]
It doesn't work well in swagger-ui (for the keyword`anyOf`). You can use `redoc` or create a new enum including `A` and `B`.

**Reply by miladva** [reply]
Can you give me an example?

**Comment by YuriiMotov** [comment]
Related discussion: https://github.com/fastapi/fastapi/discussions/12410

**Comment by YuriiMotov** [comment]
**UPD:** Seems that it works in current version of Swagger UI

<img width="480" height="270" alt="image" src="https://github.com/user-attachments/assets/70768c81-74d5-4d29-bbdf-c4435b7cde9c" />
